### PR TITLE
Robots insertion

### PR DESCRIPTION
**Pull request by BenjaminDeleze**
Allow robots in joints endpoint.

While checking that, I saw that it was also not possible to insert `robot` in 

- `fastHelix` and `slowHelix` of a propeller
- `rotatingHead` of a lidar.

But I am not sure if we want to change it.
I do not really see a utility but on the other point it is strange that:
- `fastHelix` -> `robot` is wrong
- `fastHelix` -> `solid` -> `robot` is correct

What do you think @omichel @stefaniapedrazzi ?

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
`fastHelix -> solid -> robot` is working just because when changing the rules to insert Robot nodes in Solid parent we didn't add this exception.
This is a particular case where slowHelix or fastHelix are shown but not at the same time.
So it seems quite difficult to have a consistent behavior if a Robot is inserted in these fields where it works only if the corresponding parent is active.
To me it seems that it is better to add prevent adding Robot nodes in these two fields.

**Comment by omichel** [comment]
I agree with @stefaniapedrazzi. I don't see any use-case for this and it is really odd. For the rotating head of the lidar, we could allow it if it doesn't cause any trouble, but a robot should never be added as a children or sub-children of an helix (fast or slow).

**Comment by BenjaminDeleze** [comment]
Very good point

> `fastHelix -> solid -> robot` is working just because when changing the rules to insert Robot nodes in Solid parent we didn't add this exception. This is a particular case where slowHelix or fastHelix are shown but not at the same time. So it seems quite difficult to have a consistent behavior if a Robot is inserted in these fields where it works only if the corresponding parent is active. To me it seems that it is better to add prevent adding Robot nodes in these two fields.

File: src/webots/nodes/utils/WbNodeUtilities.cpp [hunk 1]
```diff
@@ -775,6 +779,36 @@ bool WbNodeUtilities::isDescendantOfBillboard(const WbNode *node) {
...
+
+    if (baseNode->nodeType() == WB_NODE_PROPELLER)
+      return true;
+
+    field = n->parentField(true);
+    n = n->parentNode();
+  }
+
+  return false;
+}
```
Comment (stefaniapedrazzi) [inline src/webots/nodes/utils/WbNodeUtilities.cpp:810]: I see that this is copied from the `isDescendantOfBillboard` function.
But why do you need to also use the `WbNode::parentField()` (that is much more costly than just calling `WbNode::parentNode()`)?

**Review by stefaniapedrazzi — COMMENTED** [review]
The change seems fine.
I just have a comment about the code for checking the parent nodes type.

File: src/webots/nodes/utils/WbNodeUtilities.cpp (hunk 1 above)
Comment (BenjaminDeleze) [reply src/webots/nodes/utils/WbNodeUtilities.cpp]: No idea, my guess is that I took example on `WbVrmlNodeUtilities::isFieldDescendant` to first implement `isDescendantOfBillboard`.
I removed the mention to fields in both `isDescendantOfBillboard` and `isDescendantOfPropeller`

File: src/webots/nodes/utils/WbNodeUtilities.cpp [hunk 2]
```diff
@@ -758,17 +762,43 @@ bool WbNodeUtilities::isDescendantOfBillboard(const WbNode *node) {
...
+    WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(n);
+
+    if (!baseNode)
+      return false;
+
+    if (baseNode->nodeType() == WB_NODE_PROPELLER)
+      return true;
+
     n = n->parentNode();
   }
```
Comment (stefaniapedrazzi) [inline src/webots/nodes/utils/WbNodeUtilities.cpp:803]: You could also further simplify this:
```suggestion
  WbNode *n = node;
  while (n && !n->isWorldRoot()) {
    const WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(n);
    if (!baseNode)
      return false;
    if (baseNode->nodeType() == WB_NODE_PROPELLER)
      return true;
    
    n = n->parent();
  }
```

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**